x-pack/plugins/apm/public/components/shared/TransactionActionMenu/TransactionActionMenu.tsx

```diff
@@ -37,158 +38,194 @@ function getInfraMetricsQuery(transaction: Transaction) {
   readonly isOpen: boolean;
 }
 
-export class TransactionActionMenu extends React.Component<Props, State> {
-  public state: State = {
-    isOpen: false
-  };
-
-  public toggle = () => {
-    this.setState(state => ({ isOpen: !state.isOpen }));
-  };
-
-  public close = () => {
-    this.setState({ isOpen: false });
-  };
-
-  public getInfraActions(transaction: Transaction) {
-    const hostName = get(transaction, 'context.system.hostname');
-    const podId = get(transaction, 'kubernetes.pod.uid');
-    const containerId = get(transaction, 'container.id');
-    const pathname = '/app/infra';
-    const time = new Date(transaction['@timestamp']).getTime();
-    const infraMetricsQuery = getInfraMetricsQuery(transaction);
-
-    return [
-      {
-        icon: 'loggingApp',
-        label: 'Show pod logs',
-        target: podId,
-        hash: `/link-to/pod-logs/${podId}`,
-        query: { time }
-      },
-
-      {
-        icon: 'loggingApp',
-        label: 'Show container logs',
-        target: containerId,
-        hash: `/link-to/container-logs/${containerId}`,
-        query: { time }
-      },
-
-      {
-        icon: 'loggingApp',
-        label: 'Show host logs',
-        target: hostName,
-        hash: `/link-to/host-logs/${hostName}`,
-        query: { time }
-      },
-
-      {
-        icon: 'infraApp',
-        label: 'Show pod metrics',
-        target: podId,
-        hash: `/link-to/pod-detail/${podId}`,
-        query: infraMetricsQuery
-      },
-
-      {
-        icon: 'infraApp',
-        label: 'Show container metrics',
-        target: containerId,
-        hash: `/link-to/container-detail/${containerId}`,
-        query: infraMetricsQuery
-      },
-
-      {
-        icon: 'infraApp',
-        label: 'Show host metrics',
-        target: hostName,
-        hash: `/link-to/host-detail/${hostName}`,
-        query: infraMetricsQuery
-      }
-    ]
-      .filter(({ target }) => Boolean(target))
-      .map(({ icon, label, hash, query }, index) => {
-        const href = getKibanaHref({
-          location,
-          pathname,
-          hash,
-          query
-        });
-
-        return (
-          <EuiContextMenuItem icon={icon} href={href} key={index}>
-            <EuiFlexGroup gutterSize="s">
-              <EuiFlexItem>
-                <EuiLink>{label}</EuiLink>
-              </EuiFlexItem>
-              <EuiFlexItem grow={false}>
-                <EuiIcon type="popout" />
-              </EuiFlexItem>
-            </EuiFlexGroup>
-          </EuiContextMenuItem>
-        );
-      });
-  }
-
-  public render() {
-    const { transaction, location } = this.props;
-    return (
-      <QueryWithIndexPattern query={getDiscoverQuery(transaction)}>
-        {query => {
-          const discoverTransactionHref = getKibanaHref({
+export const TransactionActionMenu = injectI18n(
+  // tslint:disable-next-line:no-shadowed-variable
+  class TransactionActionMenu extends React.Component<Props, State> {
```

Comment:
`tslint` fails with `Shadowed name: 'TransactionActionMenu'tslint(no-shadowed-variable)`.  There is an issue https://github.com/palantir/tslint/issues/4247.
I'd like to leave a class name without defining `displayName` static property, so I disabled `no-shadowed-variable` rule.